### ISSUE TITLE
Sass @at-rule embedding issue

### ISSUE THREAD

**Issue by DanPurdy** [tonyganch/gonzales-pe#193]
There's an issue with the Sass syntax where an at-rule immediately following an at-rule on the same level (siblings) is actually interpreted as a child node.

``` sass
@function hello($value)
  @return $value

@function world($value)
  @return $value
```

```
 -> stylesheet
 | -> atrule
 | | -> atkeyword
 | | | -> ident
 | | |    "function"
 | | -> space
 | |    " "
 | | -> function
 | | | -> ident
 | | |    "hello"
 | | | -> arguments
 | | | | -> variable
 | | | | | -> ident
 | | | | |    "value"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "  "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "return"
 | | | | -> space
 | | | |    " "
 | | | | -> variable
 | | | | | -> ident
 | | | | |    "value"
 | | | | -> space
 | | | |    "\n"
 | | | -> space
 | | |    "\n"
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "function"
 | | | | -> space
 | | | |    " "
 | | | | -> function
 | | | | | -> ident
 | | | | |    "world"
 | | | | | -> arguments
 | | | | | | -> variable
 | | | | | | | -> ident
 | | | | | | |    "value"
 | | | | -> space
 | | | |    "\n"
 | | | | -> block
 | | | | | -> space
 | | | | |    "  "
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "return"
 | | | | | | -> space
 | | | | | |    " "
 | | | | | | -> variable
 | | | | | | | -> ident
 | | | | | | |    "value"
```

happens with media queries, mixins etc too so all at-rules seem equally affected such as with the following example

``` sass
@media (max-width: 900px)
  .test
    color: red

@function world($value)
  @return $value
```

**Comment by DanPurdy** [comment]
Another example just popped up

``` sass
@mixin media-breakpoint-notebook
    @media (max-width: 1280px) and (min-width: 1024px)
        @content

// $noop: 0 // hide sass-lint false error

@mixin media-breakpoint-msie-10-11
    @media (-ms-high-contrast: none), (-ms-high-contrast: active)
        @content
```

```
 -> stylesheet
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "orange-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "ff8011"
 | -> declarationDelimiter
 |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "vk-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "67bae3"
 | -> declarationDelimiter
 |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "telegram-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "61b3db"
 | -> declarationDelimiter
 |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "tw-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "7cd1f0"
 | -> declarationDelimiter
 |    "\n"
 | -> space
 |    "\n"
 | -> mixin
 | | -> atkeyword
 | | | -> ident
 | | |    "mixin"
 | | -> space
 | |    " "
 | | -> ident
 | |    "media-breakpoint-notebook"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "    "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "media"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "max-width"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> dimension
 | | | | | | -> number
 | | | | | |    "1280"
 | | | | | | -> ident
 | | | | | |    "px"
 | | | | -> space
 | | | |    " "
 | | | | -> ident
 | | | |    "and"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "min-width"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> dimension
 | | | | | | -> number
 | | | | | |    "1024"
 | | | | | | -> ident
 | | | | | |    "px"
 | | | | -> space
 | | | |    "\n"
 | | | | -> block
 | | | | | -> space
 | | | | |    "        "
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "content"
 | | | | | | -> space
 | | | | | |    "\n"
 | | | | | -> space
 | | | | |    "\n"
 | | | | | -> singlelineComment
 | | | | |    " $noop: 0 // hide sass-lint false error"
 | | | | | -> space
 | | | | |    "\n"
 | | | | | -> space
 | | | | |    "\n"
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "mixin"
 | | | | | | -> space
 | | | | | |    " "
 | | | | | | -> ident
 | | | | | |    "media-breakpoint-msie-10-11"
 | | | | | | -> space
 | | | | | |    "\n"
 | | | | | | -> block
 | | | | | | | -> space
 | | | | | | |    "    "
 | | | | | | | -> atrule
 | | | | | | | | -> atkeyword
 | | | | | | | | | -> ident
 | | | | | | | | |    "media"
 | | | | | | | | -> space
 | | | | | | | |    " "
 | | | | | | | | -> parentheses
 | | | | | | | | | -> ident
 | | | | | | | | |    "-ms-high-contrast"
 | | | | | | | | | -> operator
 | | | | | | | | |    ":"
 | | | | | | | | | -> space
 | | | | | | | | |    " "
 | | | | | | | | | -> ident
 | | | | | | | | |    "none"
 | | | | | | | | -> operator
 | | | | | | | |    ","
 | | | | | | | | -> space
 | | | | | | | |    " "
 | | | | | | | | -> parentheses
 | | | | | | | | | -> ident
 | | | | | | | | |    "-ms-high-contrast"
 | | | | | | | | | -> operator
 | | | | | | | | |    ":"
 | | | | | | | | | -> space
 | | | | | | | | |    " "
 | | | | | | | | | -> ident
 | | | | | | | | |    "active"
 | | | | | | | | -> space
 | | | | | | | |    "\n"
 | | | | | | | | -> block
 | | | | | | | | | -> space
 | | | | | | | | |    "        "
 | | | | | | | | | -> atrule
 | | | | | | | | | | -> atkeyword
 | | | | | | | | | | | -> ident
 | | | | | | | | | | |    "content"
```

If you uncomment the line then normal service is resumed!

``` sass
@mixin media-breakpoint-notebook
    @media (max-width: 1280px) and (min-width: 1024px)
        @content

$noop: 0 // hide sass-lint false error

@mixin media-breakpoint-msie-10-11
    @media (-ms-high-contrast: none), (-ms-high-contrast: active)
        @content

```

```
 -> stylesheet
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "orange-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "ff8011"
 | -> declarationDelimiter
 |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "vk-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "67bae3"
 | -> declarationDelimiter
 |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "telegram-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "61b3db"
 | -> declarationDelimiter
 |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "tw-color"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> color
 | | |    "7cd1f0"
 | -> declarationDelimiter
 |    "\n"
 | -> space
 |    "\n"
 | -> mixin
 | | -> atkeyword
 | | | -> ident
 | | |    "mixin"
 | | -> space
 | |    " "
 | | -> ident
 | |    "media-breakpoint-notebook"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "    "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "media"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "max-width"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> dimension
 | | | | | | -> number
 | | | | | |    "1280"
 | | | | | | -> ident
 | | | | | |    "px"
 | | | | -> space
 | | | |    " "
 | | | | -> ident
 | | | |    "and"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "min-width"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> dimension
 | | | | | | -> number
 | | | | | |    "1024"
 | | | | | | -> ident
 | | | | | |    "px"
 | | | | -> space
 | | | |    "\n"
 | | | | -> block
 | | | | | -> space
 | | | | |    "        "
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "content"
 | | | | | | -> space
 | | | | | |    "\n"
 | | | | | -> space
 | | | | |    "\n"
 | -> declaration
 | | -> property
 | | | -> variable
 | | | | -> ident
 | | | |    "noop"
 | | -> propertyDelimiter
 | |    ":"
 | | -> space
 | |    " "
 | | -> value
 | | | -> number
 | | |    "0"
 | -> space
 |    " "
 | -> singlelineComment
 |    " hide sass-lint false error"
 | -> space
 |    "\n"
 | -> space
 |    "\n"
 | -> mixin
 | | -> atkeyword
 | | | -> ident
 | | |    "mixin"
 | | -> space
 | |    " "
 | | -> ident
 | |    "media-breakpoint-msie-10-11"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "    "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "media"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "-ms-high-contrast"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> ident
 | | | | |    "none"
 | | | | -> operator
 | | | |    ","
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "-ms-high-contrast"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> ident
 | | | | |    "active"
 | | | | -> space
 | | | |    "\n"
 | | | | -> block
 | | | | | -> space
 | | | | |    "        "
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "content"
```

**Comment by tonyganch** [comment]
I rewrote part which marked where blocks ended and it should have fixed similar issues.

First example:

``` sass
@function hello($value)
  @return $value

@function world($value)
  @return $value
```

<details>
<summary>

Parse tree</summary>



```
 -> stylesheet
 | -> atrule
 | | -> atkeyword
 | | | -> ident
 | | |    "function"
 | | -> space
 | |    " "
 | | -> function
 | | | -> ident
 | | |    "hello"
 | | | -> arguments
 | | | | -> variable
 | | | | | -> ident
 | | | | |    "value"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "  "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "return"
 | | | | -> space
 | | | |    " "
 | | | | -> variable
 | | | | | -> ident
 | | | | |    "value"
 | | | | -> space
 | | | |    "\n"
 | -> space
 |    "\n"
 | -> atrule
 | | -> atkeyword
 | | | -> ident
 | | |    "function"
 | | -> space
 | |    " "
 | | -> function
 | | | -> ident
 | | |    "world"
 | | | -> arguments
 | | | | -> variable
 | | | | | -> ident
 | | | | |    "value"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "  "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "return"
 | | | | -> space
 | | | |    " "
 | | | | -> variable
 | | | | | -> ident
 | | | | |    "value"
```

</details>

Second example:

``` sass
@mixin media-breakpoint-notebook
    @media (max-width: 1280px) and (min-width: 1024px)
        @content

$noop: 0 // hide sass-lint false error

@mixin media-breakpoint-msie-10-11
    @media (-ms-high-contrast: none), (-ms-high-contrast: active)
        @content
```

<details>
<summary>

Parse tree</summary>



```
 -> stylesheet
 | -> mixin
 | | -> atkeyword
 | | | -> ident
 | | |    "mixin"
 | | -> space
 | |    " "
 | | -> ident
 | |    "media-breakpoint-notebook"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "    "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "media"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "max-width"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> dimension
 | | | | | | -> number
 | | | | | |    "1280"
 | | | | | | -> ident
 | | | | | |    "px"
 | | | | -> space
 | | | |    " "
 | | | | -> ident
 | | | |    "and"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "min-width"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> dimension
 | | | | | | -> number
 | | | | | |    "1024"
 | | | | | | -> ident
 | | | | | |    "px"
 | | | | -> space
 | | | |    "\n"
 | | | | -> block
 | | | | | -> space
 | | | | |    "        "
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "content"
 | | | | | | -> space
 | | | | | |    "\n"
 | | | -> declarationDelimiter
 | | |    "\n"
 | -> singlelineComment
 |    " $noop: 0 // hide sass-lint false error"
 | -> space
 |    "\n"
 | -> space
 |    "\n"
 | -> mixin
 | | -> atkeyword
 | | | -> ident
 | | |    "mixin"
 | | -> space
 | |    " "
 | | -> ident
 | |    "media-breakpoint-msie-10-11"
 | | -> space
 | |    "\n"
 | | -> block
 | | | -> space
 | | |    "    "
 | | | -> atrule
 | | | | -> atkeyword
 | | | | | -> ident
 | | | | |    "media"
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "-ms-high-contrast"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> ident
 | | | | |    "none"
 | | | | -> operator
 | | | |    ","
 | | | | -> space
 | | | |    " "
 | | | | -> parentheses
 | | | | | -> ident
 | | | | |    "-ms-high-contrast"
 | | | | | -> operator
 | | | | |    ":"
 | | | | | -> space
 | | | | |    " "
 | | | | | -> ident
 | | | | |    "active"
 | | | | -> space
 | | | |    "\n"
 | | | | -> block
 | | | | | -> space
 | | | | |    "        "
 | | | | | -> atrule
 | | | | | | -> atkeyword
 | | | | | | | -> ident
 | | | | | | |    "content"
```

</details>

**Comment by TheJaredWilcurt** [comment]
Using latest version of Gulp-Sass-Lint with this file:

![indentation](https://cloud.githubusercontent.com/assets/4629794/20281726/9c3e85b0-aa7f-11e6-8925-f4dbfd51adc9.png)

**Comment by henrahmagix** [comment]
@tonyganch May we please have a release of this fix? I've been using it with sass-lint@1.10.2 on a few codebases, and it would mean https://github.com/sasstools/sass-lint/issues/1084 can be closed.
